src/smallmap_gui.cpp

```diff
@@ -585,11 +585,18 @@ uint32_t GetSmallMapOwnerPixels(TileIndex tile, TileType t, IncludeHeightmap inc
 		case MP_VOID:     return MKCOLOUR_XXXX(PC_BLACK);
 		case MP_INDUSTRY: return MKCOLOUR_XXXX(PC_DARK_GREY);
 		case MP_HOUSE:    return MKCOLOUR_XXXX(PC_DARK_RED);
-		default:          o = GetTileOwner(tile); break;
-		/* FIXME: For MP_ROAD there are multiple owners.
-		 * GetTileOwner returns the rail owner (level crossing) resp. the owner of ROADTYPE_ROAD (normal road),
-		 * even if there are no ROADTYPE_ROAD bits on the tile.
-		 */
+		case MP_ROAD:
+			if (HasRoadTypeRoad(tile)) {
+				o = GetRoadOwner(tile, RTT_ROAD);
+			} else {
+				o = GetRoadOwner(tile, RTT_TRAM);
+				if (o == OWNER_TOWN) o = OWNER_NONE;
```

Comment:
This could use a comment explanation. Towns don't normally own tram tracks, so I am a bit confused what this is doing. 🙂

Reply:
https://github.com/OpenTTD/OpenTTD/blob/3b004c57bc6096e65013dfd337113149e08ea6d6/docs/landscape.html#L565
Though now I'm noticing that `GetRoadOwner` already does this, so this code isn't actually doing anything.